common/util.go

```diff
@@ -159,8 +163,33 @@ func GetDataLoc() (string, error) {
 	return GetLoc(data_dir)
 }
 
+func getAbsPath(subdir string) (dir string, err error) {
+	switch true {
+	case strings.Contains(subdir, "etc"):
+		dir = abs_etc_dir
+		break
+	case strings.Contains(subdir, "data"):
+		dir = abs_data_dir
+		break
+	case strings.Contains(subdir, "log"):
+		dir = abs_log_dir
+		break
+	case strings.Contains(subdir, "plugins"):
+		dir = abs_plugin_dir
+		break
+	}
+	if _, err = os.Stat(dir); os.IsExist(err) {
+		return dir, nil
+	}
+	return "", err
+}
+
 func GetLoc(subdir string) (string, error) {
-	dir, err := os.Getwd()
+	dir, err := getAbsPath(subdir)
```

Comment:
Need to check with @zhanghongtong which is the highest priority: absolute or relative path